### PR TITLE
Update UserManager to normalize role name

### PR DESCRIPTION
**Pull request by HaoK**
As part of this fixed the in memory test implementation to verify the fix so its touching way more files as a result

cc @divega

### CODE REVIEW

**Comment by HaoK** [comment]
Fixes https://github.com/aspnet/Identity/issues/716

File: src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs [hunk 1]
```diff
@@ -408,27 +408,27 @@ public virtual Task<bool> HasPasswordAsync(TUser user, CancellationToken cancell
...
                 throw new ArgumentNullException(nameof(user));
             }
-            if (string.IsNullOrWhiteSpace(roleName))
+            if (string.IsNullOrWhiteSpace(normalizedRoleName))
             {
-                throw new ArgumentException(Resources.ValueCannotBeNullOrEmpty, nameof(roleName));
+                throw new ArgumentException(Resources.ValueCannotBeNullOrEmpty, nameof(normalizedRoleName));
             }
-            var roleEntity = await Roles.SingleOrDefaultAsync(r => r.Name.ToUpper() == roleName.ToUpper(), cancellationToken);
+            var roleEntity = await Roles.SingleOrDefaultAsync(r => r.NormalizedName == normalizedRoleName.ToUpper(), cancellationToken);
```
Comment (HaoK) [inline src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs]: remove to upper

File: src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs [hunk 2]
```diff
@@ -436,11 +436,11 @@ public virtual Task<bool> HasPasswordAsync(TUser user, CancellationToken cancell
...
```
Comment (HaoK) [inline src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs]: remove toupper

File: src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs [hunk 3]
```diff
@@ -489,11 +489,11 @@ where userRole.UserId.Equals(userId)
...
                 throw new ArgumentNullException(nameof(user));
             }
-            if (string.IsNullOrWhiteSpace(roleName))
+            if (string.IsNullOrWhiteSpace(normalizedRoleName))
             {
-                throw new ArgumentException(Resources.ValueCannotBeNullOrEmpty, nameof(roleName));
+                throw new ArgumentException(Resources.ValueCannotBeNullOrEmpty, nameof(normalizedRoleName));
             }
-            var role = await Roles.SingleOrDefaultAsync(r => r.Name.ToUpper() == roleName.ToUpper(), cancellationToken);
+            var role = await Roles.SingleOrDefaultAsync(r => r.NormalizedName == normalizedRoleName.ToUpper(), cancellationToken);
```
Comment (HaoK) [inline src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs]: remove to upper

File: src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs [hunk 4]
```diff
@@ -1168,21 +1168,21 @@ public virtual Task<bool> GetTwoFactorEnabledAsync(TUser user, CancellationToken
...
             ThrowIfDisposed();
-            if (String.IsNullOrEmpty(roleName))
+            if (String.IsNullOrEmpty(normalizedRoleName))
             {
-                throw new ArgumentNullException(nameof(roleName));
+                throw new ArgumentNullException(nameof(normalizedRoleName));
             }
 
-            var role = await Roles.Where(x => x.Name.Equals(roleName)).FirstOrDefaultAsync(cancellationToken);
+            var role = await Roles.Where(x => x.NormalizedName.Equals(normalizedRoleName)).FirstOrDefaultAsync(cancellationToken);
```
Comment (HaoK) [inline src/Microsoft.AspNetCore.Identity.EntityFrameworkCore/UserStore.cs]: ==

**Comment by divega** [comment]
:shipit: after addressing the feedback.

**Comment by divega** [comment]
Feedback is addressed so :shipit:

**Comment by HaoK** [comment]
7ba61d6c6f6ce4eaee4c5018e71fa374d99bc868

**Pull request closed without merging**